packages/graphql-server/src/functions/graphql.ts

```diff
@@ -263,25 +269,30 @@ const useRedwoodAuthContext = (
   }
 }
 
-export const useUserContext = (
-  userContextBuilder: NonNullable<GraphQLHandlerOptions['context']>
+/**
+ * This Envelop plugin enriches the context on a per-request basis
+ * by populating it with the results of a custom function
+ * @returns
+ */
+export const usePopulateContext = (
+  populateContextBuilder: NonNullable<GraphQLHandlerOptions['context']>
```

Comment:
I think the type here isn't correct, based on the code below. it could be an async function or just a record right?

Reply:
Good question. This was something Dotan added as part of the original envelop conversion. I'll have a second look.

I have confirmed with the context/load tests that this plugin to set the context does work as expected.

Reply:
I assume that by having `NonNullable<GraphQLHandlerOptions['context']>` that refers to:


```
  /**
   * @description  Modify the resolver and global context.
   */
  context?: Context | ContextFunction
```

Meaning whatever Context or ContextFunction is and since

```
export type ContextFunction = (...args: any[]) => Context | Promise<Context>
```

then the Promise here handles the async case?

Does that appear to be the case @dac09 ?

Reply:
Or should the type simply be

```js
Context | Promise<Context>
```

Reply:
Ah I see - yeah that makes sense. No need to change I think!